PWGCF/GenericFramework/Tasks/flowGenericFramework.cxx

```diff
@@ -0,0 +1,445 @@
+        bool WithinPtRef = (ptreflow < particle.pt()) && (particle.pt() < ptrefup); // within RF pT range
+        if (WithinPtRef)
+          fGFW->Fill(particle.eta(), fPtAxis->FindBin(particle.pt()) - 1, particle.phi(), 1, 1);
+        if (WithinPtPOI)
+          fGFW->Fill(particle.eta(), fPtAxis->FindBin(particle.pt()) - 1, particle.phi(), 1, 2);
+        if (WithinPtPOI && WithinPtRef)
+          fGFW->Fill(particle.eta(), fPtAxis->FindBin(particle.pt()) - 1, particle.phi(), 1, 4);
+        if (cfgFillQA) {
+          QAregistry.fill(HIST("phi_gen"), particle.phi());
+          QAregistry.fill(HIST("eta_gen"), particle.eta());
+          QAregistry.fill(HIST("pt_gen"), particle.pt());
+        }
+      }
+      for (uint l_ind = 0; l_ind < corrconfigs.size(); l_ind++) {
+        FillFC(fFC_gen, corrconfigs.at(l_ind), centrality, l_Random);
+      }
+    }
+  }
+  PROCESS_SWITCH(GenericFramework, processGen, "Process analysis for MC generated events", false);
+
+  Filter dcaFilter = nabs(aod::track::dcaXY) < 0.2f;
+  void processReco(soa::Filtered<soa::Join<aod::Collisions, aod::CentFT0Cs>>::iterator const& collision, soa::Filtered<soa::Join<aod::Tracks, aod::TracksExtra, aod::TracksDCA, aod::McTrackLabels>> const& tracks, aod::McParticles const&)
+  {
+    float centrality = collision.centFT0C();
+    if (tracks.size() < 1)
+      return;
+    if (centrality < centbinning.front() || centrality > centbinning.back())
+      return;
+    float vtxz = collision.posZ();
+    if (cfgFillQA) {
+      QAregistry.fill(HIST("vtxZ_reco"), vtxz);
+    }
+    fGFW->Clear();
+    float l_Random = fRndm->Rndm();
+
+    for (auto& track : tracks) {
+      if (track.has_mcParticle()) {
+        auto particle = track.mcParticle();
+        if (!particle.isPhysicalPrimary() || particle.eta() < etalow || particle.eta() > etaup || particle.pt() < ptlow || particle.pt() > ptup)
+          continue;
+        bool WithinPtPOI = (ptpoilow < particle.pt()) && (particle.pt() < ptpoiup); // within POI pT range
+        bool WithinPtRef = (ptreflow < particle.pt()) && (particle.pt() < ptrefup); // within RF pT range
+        if (WithinPtRef)
+          fGFW->Fill(particle.eta(), fPtAxis->FindBin(particle.pt()) - 1, particle.phi(), 1, 1);
+        if (WithinPtPOI)
+          fGFW->Fill(particle.eta(), fPtAxis->FindBin(particle.pt()) - 1, particle.phi(), 1, 2);
+        if (WithinPtPOI && WithinPtRef)
+          fGFW->Fill(particle.eta(), fPtAxis->FindBin(particle.pt()) - 1, particle.phi(), 1, 4);
+        if (cfgFillQA) {
+          QAregistry.fill(HIST("phi_reco"), particle.phi());
+          QAregistry.fill(HIST("eta_reco"), particle.eta());
+          QAregistry.fill(HIST("pt_reco"), particle.pt());
+        }
+      }
+    }
+    for (uint l_ind = 0; l_ind < corrconfigs.size(); l_ind++) {
+      FillFC(fFC_reco, corrconfigs.at(l_ind), centrality, l_Random);
+    }
+  }
```

Comment:
In principle your MC reco processing should not differ from your data processing
The only difference should be the signature of your process function and the fact that in the case of the MC reco you check for potentially associated `mcparticle` 

For your next iteration consider the following construction for your `processData` function 
```
template <typename TrackObject>
inline int8_t ProcessTrack(TrackObject const& track)
{
  /* your processing for track either data or MC reco */

  if constexpr (framework::has_type_v<aod::mctracklabel::McParticleId, typename TrackObject::all_columns>) {
    if (track.mcParticleId() < 0) {
      /* no label!! process accordingly */
    } else {
      /* do your specific MC reco processing */
    }
  }

  /* your additional processing for track either data or MC reco */
```

What is inside the if `constexpr () {...}` is handled by the compiler in the appropriate way. In this way you don't need to replicate your code for data and for MC reco. Replicating code is always error prone! 

You can obtain a similar effect with your `datatype` template parameter